sdk/search/azure-search/samples/async_samples/sample_authentication_async.py

```diff
@@ -0,0 +1,44 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: sample_authentication.py
+DESCRIPTION:
+    This sample demonstrates how to authenticate with the Azure Congnitive Search
+    service with an API key. See more details about authentication here:
+    https://docs.microsoft.com/en-us/azure/search/search-security-api-keys
+USAGE:
+    python sample_authentication.py
+    Set the environment variables with your own values before running the sample:
+    1) AZURE_SEARCH_SERVICE_NAME - the name of your Azure Cognitive Search service
+    2) AZURE_SEARCH_INDEX_NAME - the name of your search index (e.g. "hotels-sample-index")
+    3) AZURE_SEARCH_API_KEY - your search API key
+"""
+
+import asyncio
+import os
+
+async def authentication_with_api_key_credential_async():
+    # [START create_search_client_with_key_async]
+    from azure.search.aio import SearchIndexClient
+    from azure.search import SearchApiKeyCredential
+    service_name = os.getenv("AZURE_SEARCH_SERVICE_NAME")
+    index_name = os.getenv("AZURE_SEARCH_INDEX_NAME")
+    key = os.getenv("AZURE_SEARCH_API_KEY")
+
+    search_client = SearchIndexClient(service_name, index_name, SearchApiKeyCredential(key))
```

Comment:
For cognitive, we are currently only allowing API key auth with the credential type. Because taking a simple string is an additive feature, we can change later if there is a strong motivation.

Reply:
> @johanste ? My recollection was that it was decided to prefer explicit credential objects

Yes. Two reasons. 1) many services eventually end up with multiple interpretations of what the string represents (api key, bearer token, sas key etc.) - so even though there is only one interpretation today, we prefer a specific credential type and 2) there is a requirement to be able to roll credentials.

And, as @kristapratico called out, we can add support for a string later (removing it is a breaking change)